kernel/poll.c

```diff
@@ -86,6 +86,11 @@ static inline bool is_condition_met(struct k_poll_event *event, uint32_t *state)
 			return true;
 		}
 		break;
+	case K_POLL_TYPE_PIPE_DATA_AVAILABLE:
+		if (k_pipe_read_avail(event->pipe)) {
+			*state = K_POLL_STATE_PIPE_DATA_AVAILABLE;
+			return true;
+		}
```

Comment:
There has been a recent change to the pipes implementation. If pipes are to be used, then the project configuration file (prj.conf) should specify CONFIG_PIPES=y. One implication to this change is that features that use pipes should either automatically pull in pipes or put a #ifdef CONFIG_PIPES/#endif pair around the relevant code.

For the polling, I would suggest using the #ifndef CONFIG_PIPES/#endif method. This should also resolve your CI build errors.